source/Octopus.Versioning/Octopus/OctopusVersionParser.cs

```diff
@@ -40,13 +35,10 @@ public OctopusVersion Parse(string? version)
         {
             try
             {
-                if ((version?.Trim() ?? string.Empty) == string.Empty)
+                if (string.IsNullOrWhiteSpace(version))
                     throw new ArgumentException("The version can not be an empty string");
 
-                if (version?.Contains(" ") ?? false)
-                    throw new ArgumentException("The version can not contain spaces");
-
-                var result = VersionRegex.Match(version ?? string.Empty);
+                var result = VersionRegex.Match((version ?? string.Empty).Trim());
```

Comment:
I don't really agree with this change, to me is the consumer of this API that needs to ensure whitespaces are trimmed.

Reply:
That is a fair point. The primary client here is the Octopus server, and during testing we found both the web ui and cli do strip spaces on version strings.

The change here was to retain compatibility with the existing parsing rules to make the class more robust along the lines of "be conservative in what you send, be liberal in what you accept".